### ISSUE TITLE
Question about the feature split in model forward

### ISSUE THREAD

**Issue by cocoshe** [hustvl/EVF-SAM#23]
https://github.com/hustvl/EVF-SAM/blob/781bee05fd94e6bb59c478bab44e6c51bf7959e5/model/evf_sam.py#L179

I noticed the forward part, why split the early fused text feature into many parts?

**Comment by CoderZhangYx** [comment]
Within a single image of the training dataset, there are several (usually more than 1) referring expressions corresponding to different parts of the image. For example we use batch 2 to train the code, and the first image has 3 referring expressions and the secode image has 2 referring expressions, the offset would be [0, 3, 5]. The `torch.split` would split the multi-modal extracted feat of length 5 to a list, where each item of the list corresponds to each image in batch.

**Comment by cocoshe** [comment]
> Within a single image of the training dataset, there are several (usually more than 1) referring expressions corresponding to different parts of the image. For example we use batch 2 to train the code, and the first image has 3 referring expressions and the secode image has 2 referring expressions, the offset would be [0, 3, 5]. The `torch.split` would split the multi-modal extracted feat of length 5 to a list, where each item of the list corresponds to each image in batch.

OK I get it, thanks.

And another question about the SAM2 training. When inference, we only need to give the text prompt to the first image, then SAM2 can propagate the mask to all the frames in the video. However, during training, is the loss calculated frame by frame similar to SAM1, and not using the video propagate ability?(Which means we consider the SAM2 a simple image segmenter instead of video segmenter)

So I am curious about if the video propagate ability is used during SAM2 training?

**Comment by CoderZhangYx** [comment]
We only train the image prediction capability of SAM-2. The video prediction capability comes from zero-shot.

**Comment by cocoshe** [comment]
> We only train the image prediction capability of SAM-2. The video prediction capability comes from zero-shot.

Thanks for your reply, that helps a lot~
